106-auto-rebalance-on-imbalanced-clusters.md

```diff
@@ -0,0 +1,583 @@
+The notifier will also update the ConfigMap with a `unfixableGoals` list and put the goal which cannot be fixed by rebalancing there so that the operator can then set a warning condition based on the unfixable anomaly in the status of the `Kafka` CR
+
+```yaml
+status:
+  autoRebalance:
+    state: Idle 
+    lastTransitionTime: "2025-09-22T16:04:00Z"
+    modes:
+      - mode: imbalance
+  conditions:
+    - type: Warning
+      status: "True"
+      reason: UnfixableViolatedGoal
+      message: |
+        The detected `DiskDistributionGoal` cannot be fixed.
+```
+
+#### Metrics for tracking the rebalance requests
+
+If the users want to track when the auto-rebalances happen, they can access the Strimzi [metrics](https://github.com/strimzi/strimzi-kafka-operator/blob/main/examples/metrics/grafana-dashboards/strimzi-operators.json#L712) for the `KafkaRebalance` custom resources, this includes when they were visible/created.
+These metrics also cover the `KafkaRebalance`(s) which were created automatically, so the users can utilize them to understand when an auto-rebalance were triggered in their cluster.
+
+### Auto-rebalancing execution for `imbalance` mode
+
+#### Auto-rebalancing Finite State Machine (FSM) for `imbalance` mode
+
+Currently, the auto-rebalancing mechanism runs through a Finite State Machine (FSM) made by the following states:
+
+* **Idle**: Initial state with a new auto-rebalancing initiated when scaling down/up operations were requested. This is also the ending state after an auto-rebalancing completed successfully or failed.
+* **RebalanceOnScaleDown**: a rebalancing related to a scale down operation is running.
+* **RebalanceOnScaleUp**: a rebalancing related to a scale up operation is running.
+
+With the new `imbalance` mode, we will be introducing a new state to the FSM called `RebalanceOnImbalance`.
+This state will be associated with rebalances triggered by imbalanced cluster.
+
+With the new `imbalance` mode, the FSM state transitions would look something like this:
+
+```mermaid
+flowchart TB
+  A[Idle] --scaleDown requested--> B[RebalanceOnScaleDown]
+  A[Idle] --scaleUp requested --> C[RebalanceOnScaleUp] 
+  A[Idle] --imbalanced cluster detected --> D[RebalanceOnImbalance]
+```
+* from **Idle** to:
+  * **RebalanceOnScaleDown**: if a scale down operation was requested. This transition happens even if a scale up was requested at the same time but the rebalancing on scaling down has the precedence. The rebalancing on scale up is queued. They will run sequentially.
+  * **RebalanceOnScaleUp**: if only a scale up operation was requested. There was no scale down operation requested.
+  * **RebalanceOnImbalance**: if a ConfigMap related to goal violation was detected, and the `anomalyList` is not empty
```

Comment:
I think this sentence needs to cover a similar information like the other 2 above in term of which operation has the precedence, if other operations requested at the same time. I understand the diagram tries to explain it but without a little explanation here, it's not easy to understand.